networkx/algorithms/tree/branchings.py

```diff
@@ -600,37 +616,50 @@ def is_root(G, u, edgekeys):
         for edgekey in edges:
             u, v, d = self.graphs[0].edge_index[edgekey]
             dd = {self.attr: self.trans(d[self.attr])}
-            # TODO: make this preserve the key. In fact, make this use the
-            # same edge attributes as the original graph.
+
+            # Optionally, preserve the other edge attributes of the original
+            # graph
+            if preserve_attrs:
+                for (key, value) in d.items():
+                    if key not in [self.attr, self.candidate_attr]:
+                        dd[key] = value
+
+            # TODO: make this preserve the key.
             H.add_edge(u, v, **dd)
 
         return H
 
 
-def maximum_branching(G, attr='weight', default=1):
+def maximum_branching(G, attr='weight', default=1, preserve_attrs=False):
     ed = Edmonds(G)
-    B = ed.find_optimum(attr, default, kind='max', style='branching')
+    B = ed.find_optimum(attr, default, preserve_attrs,
+                        kind='max', style='branching')
     return B
 
 
-def minimum_branching(G, attr='weight', default=1):
+def minimum_branching(G, attr='weight', default=1, preserve_attrs=False):
     ed = Edmonds(G)
-    B = ed.find_optimum(attr, default, kind='min', style='branching')
+    B = ed.find_optimum(attr, default, preserve_attrs,
+                        kind='min', style='branching')
```

Comment:
Existing users who wrote code like this:
```
ed.find_optimum(attr, default, kind, style)
```
will be broken by the placement of `preserve_attrs` before `kind` and `style` in the function signature of `find_optimum`. Since the API is not using keyword-only arguments, I'd recommend placing `preserve_attrs` last in the argument list in the definition of `find_optimum`.

Reply:
Thats a good point - I had thought the Edmonds object was generally internal-only, so happy to make the change